### PR TITLE
Add list command

### LINKED ISSUE #107: Add PropertyBookStorage, JsonAdaptedProperty, JsonPropertyBookStorage, JsonSerializablePropertyBook classes to support Property Storage
To support property storage to Json file

### LINKED ISSUE #108: Add PropertyCard, PropertyListPanel and their respective fxml classes
To support property display in Ui

### LINKED ISSUE #109: Update Storage, StorageManager,ModelManager, Userprefs to support Property Storage.
To support property storage.

### LINKED ISSUE #110: Add PropertyBookBuilder class to support testing in ModelManagerTest.
Update ModelManagerTest for property book testing.

### PR DESCRIPTION
**Pull request by candyhy**
Add PropertyBookStorage, JsonAdaptedProperty, JsonPropertyBookStorage, JsonSerializablePropertyBook classes to support Property Storage.
Add PropertyCard, PropertyListPanel and their respective fxml to support property display in Ui.
Update Storage, StorageManager,ModelManager, Userprefs to support Property Storage.
Fix bug in parsing client data in AddPropertyCommandParser.
Add PropertyBookBuilder class to support testing in ModelManagerTest.

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedProperty.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import seedu.address.model.property.client.Client;
+import seedu.address.model.remark.Remark;
+import seedu.address.model.util.DateTimeFormat;
+
+/**
+ * Jackson-friendly version of {@link Property}.
+ */
+class JsonAdaptedProperty {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Appointment's %s field is missing!";
```
Comment (Cp-John) [inline src/main/java/seedu/address/storage/JsonAdaptedProperty.java]: Should `"Appointment's %s field is missing!"` be `"Property's %s field is missing!"`?

File: src/main/java/seedu/address/storage/JsonSerializablePropertyBook.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ */
+@JsonRootName(value = "propertybook")
+class JsonSerializablePropertyBook {
+
+    public static final String MESSAGE_DUPLICATE_PROPERTY = "Property list contains duplicate properties.";
+
+    private final List<JsonAdaptedProperty> properties = new ArrayList<>();
+
+    /**
+     * Constructs a {@code JsonSerializablePropertyBook} with the given appointments.
```
Comment (Cp-John) [inline src/main/java/seedu/address/storage/JsonSerializablePropertyBook.java]: Should update the comment to `properties`

File: src/main/java/seedu/address/MainApp.java [hunk 3]
```diff
@@ -80,27 +85,47 @@ private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
...
             if (!appointmentBookOptional.isPresent()) {
-                logger.info("Data file not found. Will be starting with a sample AddressBook");
+                logger.info("Data file not found. Will be starting with a sample AppointmentBook");
             }
             //initialData = addressBookOptional.orElseGet(SampleDataUtil::getSampleAddressBook);
-            initialData = appointmentBookOptional.orElseGet(SampleDataUtil::getSampleAppointmentBook);
+            initialAppointmentData = appointmentBookOptional.orElseGet(SampleDataUtil::getSampleAppointmentBook);
         } catch (DataConversionException e) {
-            logger.warning("Data file not in the correct format. Will be starting with an empty AddressBook");
+            logger.warning("Data file not in the correct format. Will be starting with an empty Appointment");
```
Comment (Cp-John) [inline src/main/java/seedu/address/MainApp.java]: Should `empty Appointment` be `empty AppointmentBook`?

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -89,9 +90,29 @@ public ModelManager(ReadOnlyPropertyBook propertyBook, ReadOnlyUserPrefs userPre
...
 
+    /**
+     * Initializes a ModelManager with the given appointmentBook, propertyBook and userPrefs.
+     */
+    public ModelManager(ReadOnlyAppointmentBook appointmentBook, ReadOnlyPropertyBook propertyBook,
+                        ReadOnlyUserPrefs userPrefs) {
+        super();
+        requireAllNonNull(appointmentBook, propertyBook, userPrefs);
+
+        logger.fine("Initializing with address book: " + propertyBook + " and user prefs " + userPrefs);
```
Comment (Cp-John) [inline src/main/java/seedu/address/model/ModelManager.java]: Should update `address book` to `property book`?

File: src/main/java/seedu/address/storage/JsonAdaptedProperty.java (hunk 1 above)
Comment (candyhy) [reply src/main/java/seedu/address/storage/JsonAdaptedProperty.java]: Fixed, thanks!

File: src/main/java/seedu/address/storage/JsonSerializablePropertyBook.java (hunk 2 above)
Comment (candyhy) [reply src/main/java/seedu/address/storage/JsonSerializablePropertyBook.java]: fixed

File: src/main/java/seedu/address/MainApp.java (hunk 3 above)
Comment (candyhy) [reply src/main/java/seedu/address/MainApp.java]: done

File: src/main/java/seedu/address/model/ModelManager.java (hunk 4 above)
Comment (candyhy) [reply src/main/java/seedu/address/model/ModelManager.java]: Fixed as well:)

File: src/main/java/seedu/address/storage/JsonSerializablePropertyBook.java (hunk 2 above)
Comment (dvdweien) [reply src/main/java/seedu/address/storage/JsonSerializablePropertyBook.java]: i think he is referring to line 27, the last word before the full stop

**Review by dvdweien — APPROVED** [review]
LGTM

**Pull request merged**